Don't unconditionally compile runJITCPPTests

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31162 Don't use RTLD_GLOBAL to load _C.
* #31161 Uniformly apply Windows logic in cpp_extensions everywhere
* **#31236 Don't unconditionally compile runJITCPPTests**

It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262581](https://our.internmc.facebook.com/intern/diff/D19262581)

ezyang committed:
Don't unconditionally compile runJITCPPTests

It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Don't unconditionally compile runJITCPPTests"


It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit b16794c:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b16794ccae79ec3d34f0a744baba4071c6db244f).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3945592) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_test (1/1)
**Step:** "Run in docker" ([details](https://dr.pytorch.org/build-details.html?build_id=73940646))
```
c57b55f42599: Pull complete 
10026a3c61bf: Pull complete 
5323f5e66f35: Pull complete 
f6a256c8ebf8: Pull complete 
4ba92bf1c34f: Pull complete 
4bdea3900720: Pull complete 
496c7bf08704: Pull complete 
e5606baa75ee: Pull complete 
3a465fc337db: Pull complete 
ca9f722afd03: Pull complete 
docker: failed to register layer: Error processing tar file(exit status 1): unexpected EOF. 
See 'docker run --help'. 
+ export id= 
+ id= 
+ docker cp /home/circleci/project/. :/circleci_stuff 
must specify at least one container source 
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang committed:
Update on "Don't unconditionally compile runJITCPPTests"


It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

eellison reviewed (approved):
LGTM, maybe add note https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L760 to cross-reference torch/CMakeLists.txt

ezyang committed:
Update on "Don't unconditionally compile runJITCPPTests"


It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262581](https://our.internmc.facebook.com/intern/diff/D19262581)

[ghstack-poisoned]

ezyang committed:
Update on "Don't unconditionally compile runJITCPPTests"


It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262581](https://our.internmc.facebook.com/intern/diff/D19262581)

[ghstack-poisoned]

ezyang committed:
Update on "Don't unconditionally compile runJITCPPTests"


It is not compiled on Windows

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262581](https://our.internmc.facebook.com/intern/diff/D19262581)

[ghstack-poisoned]
